### PR TITLE
feat: add `**kwargs` support to `XGBClassifier` and `XGBRegressor`

### PR DESCRIPTION
**Pull request by tswast**
This is in alignment with https://xgboost.readthedocs.io/en/stable/python/python_api.html\#xgboost.XGBRegressor if considering BQML to be a booster type.

Thank you for opening a Pull Request! Before submitting your PR, there are a few things you can do to make sure it goes smoothly:
- [ ] Make sure to open an issue as a [bug/issue](https://github.com/googleapis/python-bigquery-dataframes/issues/new/choose) before writing your code!  That way we can discuss the change, evaluate designs, and agree on the general idea
- [ ] Ensure the tests and linter pass
- [ ] Code coverage does not decrease (if any source code was changed)
- [ ] Appropriate docs were updated (if necessary)

Fixes internal customer request 🦕

### CODE REVIEW

File: bigframes/ml/ensemble.py [hunk 1]
```diff
@@ -139,6 +141,8 @@ def _bqml_options(self) -> Dict[str, str | int | bool | float | List[str]]:
             "enable_global_explain": self.enable_global_explain,
             "xgboost_version": self.xgboost_version,
         }
+        options.update(self._extra_bqml_options)
```
Comment (GarrettWu) [inline bigframes/ml/ensemble.py:144]: It may override existing options, if user pass in both.

We use sklearn terms instead of BQML terms for existing parameters, e.g. learning_rate instead of LEARN_RATE. User may get confused looking at two sets of options at the same time.

It becomes hard for us to both stick with sklearn-like experience and support BQML other offerings. 

Or maybe we can ask them to only use one set of parameters, and raise error of conflicts?

File: tests/unit/ml/test_golden_sql.py [hunk 2]
```diff
@@ -286,3 +287,83 @@ def test_decomposition_mf_score_with_x(mock_session, bqml_model, mock_X):
...
+              max_tree_depth=6,
+              subsample=1.0,
+              l1_reg=0.0,
+              l2_reg=1.0,
+              learn_rate=0.3,
+              max_iterations=20,
+              min_rel_progress=0.01,
+              enable_global_explain=False,
+              xgboost_version='0.9',
+              category_encoding_method='LABEL_ENCODING',
```
Comment (GarrettWu) [inline tests/unit/ml/test_golden_sql.py:324]: Do they want to use label encoding on features? This is actually wrong. https://stackoverflow.com/a/34346937

I talked with jiashangliu@, but BQML won't change the default.

**Comment by tswast** [comment]
Closing. I think the alternative of exposing the SQL methods directly in the `bigframes.bigquery.ml` module, similar to `bigframes.bigquery.ai` will be more useful.
